source/ut_metadata.pkb

```diff
@@ -102,319 +111,37 @@ create or replace package body ut_metadata as
-    if l_package_comments is not null then
-      a_annotated_pkg.annotations := get_annotations(l_package_comments);
-    end if;
+  end get_package_spec_source;
 
-    -- loop through procedures and functions of the package and get all the comment blocks just before it's declaration
-    l_annot_proc_ind := 1;
-    loop
-      l_annot_proc_ind := regexp_instr(srcstr     => l_pkg_spec
-                                      ,pattern    => c_annotation_block_pattern
-                                      ,occurrence => 1
-                                      ,modifier   => 'i'
-                                      ,position   => l_annot_proc_ind);
-
-      exit when l_annot_proc_ind = 0;
-
-      l_annot_proc_block := regexp_substr(srcstr     => l_pkg_spec
-                                         ,pattern    => c_annotation_block_pattern
-                                         ,position   => l_annot_proc_ind
-                                         ,occurrence => 1
-                                         ,modifier   => 'i');
-
-      l_annot_proc_ind := l_annot_proc_ind + length(l_annot_proc_block);
-
-      l_proc_comments := trim(regexp_substr(srcstr        => l_annot_proc_block
-                                           ,pattern       => c_annotation_block_pattern
-                                           ,modifier      => 'i'
-                                           ,subexpression => 1));
-      l_proc_name     := trim(regexp_substr(srcstr        => l_annot_proc_block
-                                           ,pattern       => c_annotation_block_pattern
-                                           ,modifier      => 'i'
-                                           ,subexpression => 4));
-
-      -- parse the comment block for the syntactically correct annotations and store them as an array
-      a_annotated_pkg.procedures(l_proc_name) := get_annotations(l_proc_comments);
-
-    end loop;
-
-    -- printing out parsed structure for debugging
-    $if $$ut_trace $then
-    print_parse_results;
-    $end
-
-  end parse_package_annotations;
-
-  function get_annotation_param(a_param_list tt_annotation_params, a_def_index pls_integer) return varchar2 is
-    l_result varchar2(32767);
-  begin
-    if a_param_list.exists(a_def_index) then
-      l_result := a_param_list(a_def_index).value;
-    end if;
-    return l_result;
-  end get_annotation_param;
 begin
   declare
-    l_cur_temp sys_refcursor;
+    l_cursor sys_refcursor;
+    l_object_does_not_exist exception;
+    pragma exception_init (l_object_does_not_exist, -942);
   begin
-
-    l_cur_temp     := get_source_cursor('dba_source', null, null);
+    l_cursor := get_package_spec_source_cursor('dba_source');
+    close l_cursor;
     g_source_view := 'dba_source';
-    close l_cur_temp;
-
-  exception
-    when others then
-      g_source_view := 'all_source';
+   exception
+     when l_object_does_not_exist then
+       g_source_view := 'all_source';
   end;
```

Comment:
Could you please steal "lazy" implementation from my PR to your branch, because sys_refcursor causes a bug (ORA-600) on our setup. Don't merge other staff.